client/src/components/ui/inputs.tsx

```diff
@@ -313,41 +314,36 @@ export function DropdownInput({
     setExpanded(false);
     onChange(e);
   }
-
+  function handleOnClick() {
+    setExpanded(!isExpanded);
+  }
   const containerStyle =
     `duration-50 flex flex-row h-14 w-full items-center rounded-penni-border px-4 border-2 transition-all ease-out` +
     ` ${isExpanded ? selectedStyle : deselectedStyle} ` +
     ` ${expandedOrNotEmpty ? expandedStyle : collapsedStyle} `;
 
   return (
     <div className="relative h-auto w-full">
-      <div className="m-4 w-auto">
-        <button
-          id={menuId}
-          type="button"
-          className={containerStyle}
-          aria-haspopup={true}
-          aria-expanded={true}
-          onClick={() => setExpanded(!isExpanded)}
-        >
-          <div className="flex w-full flex-col items-start">
-            {label && (
-              <label
-                htmlFor={menuId}
-                className={`${value !== "" ? labelStyleSmall : labelStyleLarge} duration-50 text-left transition-all ease-out`}
-              >
-                {label}
-              </label>
-            )}
-            {value !== "" && (
-              <span className={`${valueStyle} text-left`}>{value}</span>
-            )}
-          </div>
-          <div className="ml-3 flex size-6 items-center justify-center">
-            <DropdownIcon />
-          </div>
-        </button>
-      </div>
+      <DropdownButton
+        id={menuId}
+        style={containerStyle}
+        caretWidth={24}
+        onClick={handleOnClick}
+      >
+        <div className="flex w-full flex-col items-start">
+          {label && (
+            <label
+              htmlFor={menuId}
+              className={`${value !== "" ? labelStyleSmall : labelStyleLarge} duration-50 text-left transition-all ease-out`}
+            >
+              {label}
+            </label>
+          )}
+          {value !== "" && (
+            <span className={`${valueStyle} text-left`}>{value}</span>
+          )}
+        </div>
+      </DropdownButton>
 
       {isExpanded && (
         <DropdownMenu
```

Comment:
Remove, logic belongs in `<dropdownButton>`

Reply:
`DropdownInput` component:
<img width="536" alt="image" src="https://github.com/codersforcauses/penni/assets/98962967/52d93414-83e7-4aa7-ad9b-eb765401b42f">
`MarketDropdown` component:
<img width="529" alt="image" src="https://github.com/codersforcauses/penni/assets/98962967/e5d5a3ab-b2b4-49ad-a7db-7c7c8971cae5">

They have slightly different logic because the style is different. In `DropdownInput` component, style of the `DropdownMenu` will change based on `isExpended`. It is hard to move the `isExpended` part to `DropdownButton`

Reply:
Could you highlight what style is affected by moving `isExpanded` into the button component? I can have a look if needed

Reply:
Also, please add `onBlur` to `DropdownButton` that set `isExpanded` to false so the dropdown menu closes when the user clicks away. I forgot to add this in when making the input components